packages/block-editor/src/components/contrast-checker/contrast-checker-message.js

```diff
@@ -0,0 +1,58 @@
+/**
+ * WordPress dependencies
+ */
+import { speak } from '@wordpress/a11y';
+import { __ } from '@wordpress/i18n';
+import { Notice } from '@wordpress/components';
+import { useEffect } from '@wordpress/element';
+
+export default function ContrastCheckerMessage( {
+	colordBackgroundColor,
+	colordTextColor,
+	colordLinkColor,
+	backgroundColor,
+	textColor,
+	linkColor,
+	shouldShowTransparencyWarning,
+} ) {
+	let msg = '';
+	if ( shouldShowTransparencyWarning ) {
+		msg = __( 'Transparent text may be hard for people to read.' );
+	} else {
+		const backgroundColorBrightness = colordBackgroundColor.brightness();
+		msg =
+			( colordTextColor &&
+				backgroundColorBrightness < colordTextColor.brightness() ) ||
+			( colordLinkColor &&
+				backgroundColorBrightness < colordLinkColor.brightness() )
+				? __(
+						'This color combination may be hard for people to read. Try using a darker background color and/or a brighter text color.'
+				  )
+				: __(
+						'This color combination may be hard for people to read. Try using a brighter background color and/or a darker text color.'
```

Comment:
I think the messages here could be a little clearer. I can see the reference to text color being a point of confusion when it's actually the link color triggering the warning.

Reply:
As part of the[ contrast checker message refactor](https://github.com/WordPress/gutenberg/pull/38100#pullrequestreview-861618806) proposed above, we could try to update the messages so that they discern between link and text.

I agree that more clarity is better always.

I'll see how much of the refactor makes sense for this PR, and try to extract points to follow up on.